src/lambda/commands/importLambda.ts

```diff
@@ -0,0 +1,233 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as AdmZip from 'adm-zip'
+import { Lambda } from 'aws-sdk'
+import * as fs from 'fs-extra'
+import got from 'got'
+import * as _ from 'lodash'
+import * as path from 'path'
+import { Stream } from 'stream'
+import { promisify } from 'util'
+import * as vscode from 'vscode'
+import { LaunchConfiguration, getReferencedHandlerPaths } from '../../shared/debug/launchConfiguration'
+import { ext } from '../../shared/extensionGlobals'
+import { makeTemporaryToolkitFolder, fileExists } from '../../shared/filesystemUtilities'
+import { getLogger } from '../../shared/logger'
+import { createCodeAwsSamDebugConfig } from '../../shared/sam/debugger/awsSamDebugConfiguration'
+import { ExtensionDisposableFiles } from '../../shared/utilities/disposableFiles'
+import * as pathutils from '../../shared/utilities/pathUtils'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { Window } from '../../shared/vscode/window'
+import { LambdaFunctionNode } from '../explorer/lambdaFunctionNode'
+import { getFamily, RuntimeFamily } from '../models/samLambdaRuntime'
+import { showConfirmationMessage } from '../../s3/util/messages'
+import { addWorkspaceFolder } from '../../shared/utilities/workspaceUtils'
+
+const pipeline = promisify(Stream.pipeline)
+
+export async function importLambdaCommand(functionNode: LambdaFunctionNode, window = Window.vscode()) {
+    const workspaceFolders = vscode.workspace.workspaceFolders || []
+    const labelToWorkspace = new Map(workspaceFolders.map(folder => [`$(root-folder-opened) ${folder.name}`, folder]))
+    const otherLocationName = `$(folder-opened) ${localize('AWS.lambda.import.otherLocation', 'Select a folder...')}`
+
+    // TODO: Move to FolderQuickPickItem (would have to modify BrowseFolderQuickPickItem to non-Wizard dependence)?
+    const selectedLocation = await vscode.window.showQuickPick([...labelToWorkspace.keys(), otherLocationName], {
+        placeHolder: localize('AWS.lambda.import.prompt.placeholder', 'Select the import location'),
+    })
+
+    if (!selectedLocation) {
+        getLogger().info('ImportLambda cancelled')
+        return
+    }
+
+    let selectedUri: vscode.Uri
+    const isWorkspaceFolder = selectedLocation != otherLocationName
+    if (isWorkspaceFolder) {
+        selectedUri = labelToWorkspace.get(selectedLocation)!.uri
+    } else {
+        const selectedDirectory = await window.showOpenDialog({
+            openLabel: 'Select',
+            canSelectFiles: false,
+            canSelectFolders: true,
+        })
+        if (!selectedDirectory) {
+            getLogger().info('ImportLambda cancelled')
+            return
+        }
+        selectedUri = selectedDirectory[0]
+        addWorkspaceFolder({ uri: selectedUri })
```

Comment:
Unless things have changed, if user doesn't have a workspace opened, adding a workspace folder will restart vs code and the extension, which likely kills this function. If that is the case, consider quitting with a message at the start of the command, asking user to create/open a workspace first.